EdFi.Buzz.UI.Angular/src/app/Features/UploadSurvey/uploadSurvey.tsx

```diff
@@ -0,0 +1,415 @@
+    setIsFileSelected(false);
+    setIsFileUploading(false);
+    setQueryTime(5000);
+    loadLastUploadedSurvey();
+    OnInit();
+  }, []);
+
+  function cancelUploadStatusChecking() {
+    if (jobStatusTimer) {
+      clearTimeout(jobStatusTimer);
+      setJobStatusTimer(null);
+    }
+  }
+
+  const resetFileStatusMessage = () => {
+    setFileStatusMessage(
+      {
+        fileName: '',
+        isValid: false,
+        status: '',
+        error: null,
+        jobId: null,
+        serverJobStatus: null
+      }
+    );
+  };
+
+  function createFileStatusMessage (fileStatus: FileStatus): FileStatus {
+    if (!fileStatus || !fileStatus.fileName) {
+      return fileStatusMessage;
+    }
+    const newFileStatus = fileStatus ? (
+        {
+          fileName: fileStatus.fileName,
+          isValid: fileStatus.isValid,
+          status: fileStatus.status,
+          error: fileStatus.error,
+          jobId: fileStatus.jobId,
+          serverJobStatus: fileStatus.serverJobStatus
+        }
+      ) :
+      (
+        {
+          fileName: '',
+          isValid: false,
+          status: '',
+          error: null,
+          jobId: null,
+          serverJobStatus: null
+        }
+      );
+   return newFileStatus;
+  }
+
+   const OnInit = async () => {
+    if (surveyKey && surveyKey.length > 0) {
+      const userSurveyList = await (api.survey.getSurveyStatus(
+        api.authentication.currentUserValue.teacher.staffkey, null));
+      if (userSurveyList) {
+        setSurveyList(userSurveyList);
+        setSurveyToUpdate(userSurveyList.find(el => el.surveykey === +surveyKey));
+      }
+    }
+  };
+
+  function loadLastUploadedSurvey() {
+    const message = JSON.parse(storage.getItem('lastUploadedSurvey'));
+    if (message && message.serverJobStatus && !api.survey.JOB_STATUS_FINISH_IDS.includes(message.serverJobStatus.jobstatuskey)) {
+      setFileStatusMessage(createFileStatusMessage(message));
+      GetJobStatus(message.serverJobStatus.staffkey, message.serverJobStatus.jobkey, message);
+    } else {
+      resetFileStatusMessage();
+      return undefined;
+    }
+    return message;
+  }
+
+  function saveLastUploadedSurvey(message: FileStatus) {
+    if (message && message.fileName && message.fileName.length > 0) {
+      storage.setItem('lastUploadedSurvey', JSON.stringify(message));
+    }
+  }
+
+  function CheckFileValid(file: File): FileStatus {
+    if (!file) {
+      return { fileName: '', status: 'ERROR', error: 'No file selected', isValid: false };
+    }
+    if (file.size > SURVEY_MAX_FILE_SIZE_BYTES) {
+      const error = `File size (${(file.size / 1024.0).toFixed(2)} Kb) must be less than ${(SURVEY_MAX_FILE_SIZE_BYTES / (1024)).toFixed(2)} Kb`;
+      const message = { fileName: file.name, status: 'ERROR', error: error, isValid: false };
+      return message;
+    }
+    return { fileName: file.name, status: 'VALID', isValid: true };
+  }
+
+  function prepareFilesList(files: any) {
```

Comment:
Consider using arrow syntax consistently  for functions throughout. React does use this as default because it helps guarantee scope safety, looks cleaner. Plus, it is nicer to read the same syntax everywhere.